### PR TITLE
new actions ResizeBy and ResizeTo

### PR DESCRIPTION
**Pull request by charlesstlaurent**
New actions to resize (mostly widget) instead of scaling.

### CODE REVIEW

**Comment by minggo** [comment]
could you please add a test case for it, thanks

**Comment by charlesstlaurent** [comment]
Hi @minggo,  no problem, should I put the test with the other actions (but setContentSize does nothing on a sprite so I would have to add a Widget like an ImageView for example), or should I put it with the UI tests where there already is something that I can apply the action on?

**Comment by minggo** [comment]
@charlesstlaurent so the actions only have effect on Widget? Why need these actions?

**Comment by charlesstlaurent** [comment]
Hi @minggo, it works on all nodes where setContentSize is effective (which is not the case for Sprites). But it's mostly useful for nodes where 9-slice is enabled, because if gives different results than ScaleTo/ScaleBy in that case.

**Comment by minggo** [comment]
@charlesstlaurent i think you can add test case with other actions but use widget instead.

> But it's mostly useful for nodes where 9-slice is enabled, because if gives different results than ScaleTo/ScaleBy in that case.

Could you please add comment about it too? Then developers may know how to use it correctly. Thanks.

File: cocos/2d/CCActionInterval.h [hunk 1]
```diff
@@ -768,6 +768,92 @@ class CC_DLL SkewBy : public SkewTo
...
+CC_CONSTRUCTOR_ACCESS:
+    ResizeTo() {}
+    virtual ~ResizeTo() {}
+    
+    //
+    // Overrides
+    //
+    virtual ResizeTo* clone() const override;
+    void startWithTarget(cocos2d::Node* target) override;
+    void update(float time) override;
```
Comment (minggo) [inline cocos/2d/CCActionInterval.h:791]: why put these functions in `CC_CONSTRUCTOR_ACCESS`?

**Comment by charlesstlaurent** [comment]
Hi @minggo mistakes were made. It's now fixed.

**Pull request merged**